L1Trigger/L1THGCal/test/HGCalBackendStage1ParameterExtractor.cc

```diff
@@ -0,0 +1,293 @@
+#include <iostream> // std::cout
+#include <fstream>  // std::ofstream
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/stream/EDAnalyzer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "DataFormats/ForwardDetId/interface/ForwardSubdetector.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCScintillatorDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCSiliconDetId.h"
+#include "DataFormats/ForwardDetId/interface/HFNoseTriggerDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCalTriggerDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCalTriggerModuleDetId.h"
+#include "DataFormats/ForwardDetId/interface/HGCalTriggerBackendDetId.h"
+
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerGeometryBase.h"
+
+#include <nlohmann/json.hpp>
+using json = nlohmann::ordered_json;  // using ordered_json for readability
+
+class HGCalBackendStage1ParameterExtractor : public edm::stream::EDAnalyzer<> {
+public:
+  explicit HGCalBackendStage1ParameterExtractor(const edm::ParameterSet&);
+  ~HGCalBackendStage1ParameterExtractor();
+
+  virtual void beginRun(const edm::Run&, const edm::EventSetup&);
+  virtual void analyze(const edm::Event&, const edm::EventSetup&);
+
+private:
+  void fillTriggerGeometry(json& json_file);
+
+  edm::ESHandle<HGCalTriggerGeometryBase> triggerGeometry_;
+  edm::ESGetToken<HGCalTriggerGeometryBase, CaloGeometryRecord> triggerGeomToken_;
+
+  // Metadata
+  std::string cmssw_version_;
```

Comment:
Are there some clean ways to retrieve this or should it just be added as strings to the analyzer PSet?

Reply:
It could maybe be extracted from the names of the geometry XML files which are loaded, for instance:
https://github.com/hgc-tpg/cmssw/blob/9bff0b25ff7a1ebf947c8ef0043ea00e35efb07c/Geometry/CMSCommonData/python/cmsExtendedGeometry2026D49XML_cfi.py#L62
extracting `v11`

Reply:
Now reading geometry version from the geometry xml file name. I'm not retrieving the CMSSW era though; is that needed or redundant with the geometry version?

Reply:
That's fine. It would indeed be redundant with the geometry version